File: codecamp/src/main/java/com/gdogaru/codecamp/view/main/MainViewItem.java
package com.gdogaru.codecamp.view.main;

import com.gdogaru.codecamp.model.Schedule;

/**
 * Created by Gabriel on 2/15/2017.
 */

public interface MainViewItem {

    class AgendaItem implements MainViewItem {
        private final Schedule schedule;

        public AgendaItem(Schedule schedule) {
            this.schedule = schedule;
        }

        public Schedule getSchedule() {
            return schedule;
        }
    }

    class SpeakersItem implements MainViewItem {
    }

    class SponsorsItem implements MainViewItem {
    }
}

File: codecamp/src/main/java/com/gdogaru/codecamp/view/home/MainViewItem.kt
package com.gdogaru.codecamp.view.home

import com.gdogaru.codecamp.api.model.Schedule

/**
 * Created by Gabriel on 2/15/2017.
 */

interface MainViewItem {

    data class AgendaItem(val index: Int, val schedule: Schedule) : MainViewItem

    class SpeakersItem : MainViewItem

    class SponsorsItem : MainViewItem
}
